src/server/editorServices.ts

```diff
@@ -390,6 +391,8 @@ namespace ts.server {
         public readonly cancellationToken: HostCancellationToken;
         public readonly useSingleInferredProject: boolean;
         public readonly useInferredProjectPerProjectRoot: boolean;
+        private readonly disableInferredProjectLanguageService?: boolean;
+        private readonly ignoreConfigFiles?: boolean;
         public readonly typingsInstaller: ITypingsInstaller;

```

Comment:
Personally, I think it's more readable with a collection of independent, self-documenting flags.  However, I'm okay with eliminating them in favor of a single `syntaxOnly` flag that arbitrarily tweaks behavior.

Reply:
As an example, if we decide that we don't like the `ignoreConfigFiles` behavior, reverting it will be much easier if it's controlled by a separate flag (vs determining which usages of `syntaxOnly` contribute to ignoring config files and whether or not they affect other syntax-only behaviors).

Reply:
Done.